name_that_hash/hashes.py

```diff
@@ -1641,4 +1641,59 @@ class HashInfo:
                 )
         ],
     ),
+    Prototype(
+    regex=re.compile(r"\$bitcoin\$[0-9]{2,4}\$[a-fA-F0-9$]{250,350}", re.IGNORECASE),
+    modes=[
+        HashInfo(
+            name="Bitcoin / Litecoin",
+            hashcat=11300,
+            john="bitcoin",
+            extended=False,
+            description="Use Bitcoin2John.py to extract the hash for cracking."
+            )
+        ],
+    ),
+    Prototype(
+    regex=re.compile(r"\$ethereum\$[a-z0-9*]{150,250}", re.IGNORECASE),
+    modes=[
+        HashInfo(
+            name="Ethereum PBKDF2",
+            hashcat=15600,
+            john="ethereum-opencl",
+            extended=False,
+            description="Use ethereum2john.py to crack."
+            ),
+        HashInfo(
+            name="Ethereum Presale",
+            hashcat=16300,
+            john="ethereum-presale-opencl",
+            extended=False,
+            description="Use ethereum2john.py to crack."
+        )
+        ],
+    ),
+    Prototype(
+    regex=re.compile(r"\$monero\$(0)\*[a-f0-9]{32,3196}", re.IGNORECASE),
+    modes=[
+        HashInfo(
+            name="Monero",
+            hashcat=None,
+            john="Monero",
+            extended=False,
+            description="Use monero2john.py to crack."
+            )
+        ],
+    ),
+    Prototype(
+    regex=re.compile(r"^D{1}[5-9A-HJ-NP-U]{1}[1-9A-HJ-NP-Za-km-z]{32}$", re.IGNORECASE),
```

Comment:
I've thought about it and I think we shouldn't add Dogecoin. It's not an actual hash type and there are a lot of bitcoin-based altcoins that could also be cracked using the Bitcoin hash format. If we add it then we'll have to add all the other altcoins too.